### PR TITLE
Feature/regarch 122 object card with carousel

### PR DESCRIPTION
**Pull request by igor-kamil**
screenshot-www register-architektury local-2020-09-22-23-22-35-508

### CODE REVIEW

File: resources/js/components/card-carousel.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+require('flickity-as-nav-for');
+require('flickity-fullscreen');
+
+$('.card-carousel').each(function(i, obj) {
+    const nextButton = $(obj).find('.next-button')[ 0 ];
+    const prevButton = $(obj).find('.prev-button')[ 0 ];
+
+    // Initialize main carousel
```
Comment (eronisko) [inline resources/js/components/card-carousel.js:8]:
```suggestion
```

File: resources/sass/_overflow.scss [hunk 2]
```diff
@@ -1,3 +1,6 @@
 .overflow-x-hidden {
     overflow-x: hidden;
-}
\ No newline at end of file
+}
+.overflow-y-hidden {
+    // overflow-y: hidden;
+}
```
Comment (eronisko) [inline resources/sass/_overflow.scss:6]:
```suggestion
```

File: resources/views/components/building-card.blade.php [hunk 3]
```diff
@@ -1,10 +1,11 @@
...
-        @if($building->processedImages->isNotEmpty())
-            <a href="{{ $building->url }}" class="image-link">
-                {{ $building->cover_image->img()->attributes(['class' => 'card-img-top mb-4']) }}
-            </a>
-        @endif
+        {{-- <a href="{{ $building->url }}" class="image-link"> --}}
+            @include('components.card-carousel', [
+                'images' => $building->processedImages->map(fn ($image) => $image->getFirstMedia())
+            ])
+        {{-- </a> --}}
```
Comment (eronisko) [inline resources/views/components/building-card.blade.php:7]:
```suggestion
            @include('components.card-carousel', [
                'images' => $building->processedImages->map(fn ($image) => $image->getFirstMedia())
            ])
```

File: resources/js/app.js [hunk 4]
```diff
@@ -2,6 +2,7 @@ require('./bootstrap');
 require('./components/filters');
 require('./components/buildings-carousel');
 require('./components/gallery-carousel');
+require('./components/card-carousel');
```
Comment (eronisko) [inline resources/js/app.js:5]: I'd keep this within the building-card component. Or are you foreseeing this being re-used elsewhere?

**Review by eronisko — COMMENTED** [review]
Looks good, just some nit-picks from me so far. I'd perhaps keep it in the building-card component.

I haven't checked -- is the styleguide working now?

File: resources/js/app.js (hunk 4 above)
Comment (igor-kamil) [reply resources/js/app.js:5]: I like to keep it easily reusable anywhere else. maybe not right now. but if you don't mind - I will keep it separated.

File: resources/views/components/building-card.blade.php (hunk 3 above)
Comment (igor-kamil) [reply resources/views/components/building-card.blade.php:7]: this is actually the issue, why it's opened just as draft. When wrapping the whole carousel with `<a href...></a>`, I couldn't prevent loading the detail of building after clicking the next/prev arrows. 
when the link is not present, it works nicely. but on the other hands it's bit harder to reach building detail.
what's your suggestion?

Comment (igor-kamil) [reply resources/views/components/building-card.blade.php:7]: also ... maybe we could deploy it as it is now - and it will clarify after using/testing

**Pull request merged**